src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/telemetry/LocationAttributeTelemetryEvent.kt

```diff
@@ -0,0 +1,41 @@
+package uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.telemetry
+
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.common.toIsoDate
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.entity.LocationAttribute
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.entity.LocationStatus
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.ServiceUser
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.User
+
+class LocationAttributeTelemetryEvent(private val attribute: LocationAttribute, private val raisedBy: User) : StandardTelemetryEvent("BVLS-room-status-event") {
+  init {
+    if (attribute.amendedTime == null) {
+      require(attribute.createdBy == raisedBy.username) {
+        "Cannot create location status telemetry event, created by user does not match raised by user."
+      }
+    } else {
+      require(attribute.amendedBy == raisedBy.username) {
+        "Cannot create location status telemetry event, amended by user does not match raised by user."
+      }
+    }
+  }
+
+  override fun properties(): Map<String, String> = run {
+    buildMap {
```

Comment:
I think this covers the main things.  Let me know if you think differently.

Reply:
Maybe usage?  ie.. is it Court, Probation, Shared, or Schedule, but everything else fine